### PR TITLE
Add an option to force profiling right in the beginning of premain

### PR DESCRIPTION
**Pull request by mar-kolya**
Currently this blows up JVMs before 14

### CODE REVIEW

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java [hunk 1]
```diff
@@ -46,6 +46,9 @@
 
   public static void start(final Instrumentation inst, final URL bootstrapURL) {
     createParentClassloader(bootstrapURL);
+
+    startProfilingAgent(bootstrapURL, true);
```
[the remark this answers is not in the document]
Comment (mar-kolya) [reply dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/Agent.java]: Updated

**Comment by mar-kolya** [comment]
I think everything has been updated now in terms of variable renames - please let me know if this is better.

As far as `experimental` goes - I'm not sure what would be better way to handle this. Currently this can only be used with java14 that has not been released yet. I hope that at some point those fixes will be backported and we can remove experimental thing.

**Comment by dbenamydd** [comment]
If we think we might want to change/remove it later, but about adding something into the setting name to indicate this? Eg `dd.experimental.x`.

**Comment by thegreystone** [comment]
> As far as `experimental` goes - I'm not sure what would be better way to handle this. Currently this can only be used with java14 that has not been released yet. I hope that at some point those fixes will be backported and we can remove experimental thing.

I think this is an okay way to handle it. We're working on the backports right now.

File: dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java [hunk 2]
```diff
@@ -19,13 +19,14 @@
...
    */
-  public static synchronized void run(final boolean early) throws IllegalArgumentException {
+  public static synchronized void run(final boolean isStartingFirst)
+      throws IllegalArgumentException {
     if (PROFILER == null) {
       final Config config = Config.get();
-      if (early && !config.isProfilingStartForceEarly()) {
-        log.info("Profiling: not starting early");
+      if (isStartingFirst && !config.isProfilingStartForceFirst()) {
+        log.info("Profiling: not starting first");
```
Comment (tylerbenson) [inline dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java]: debug?

Comment (mar-kolya) [reply dd-java-agent/agent-profiling/src/main/java/com/datadog/profiling/agent/ProfilingAgent.java]: fixed

**Pull request merged**